node/node.go

```diff
@@ -412,6 +410,8 @@ func (nc *Config) Build(ctx context.Context) (*Node, error) {
 		MsgQueryer:   msg.NewQueryer(nc.Repo, fcWallet, chainReader, &cstOffline, bs),
 		MsgSender:    msg.NewSender(nc.Repo, fcWallet, chainReader, msgPool, fsub.Publish),
 		MsgWaiter:    msg.NewWaiter(chainReader, bs, &cstOffline),
+		Subscriber:   ps.NewSubscriber(fsub),
+		Publisher:    ps.NewPublisher(fsub),
```

Comment:
Tell me if this is overkill and the plumbing should just have a `libp2p.pubsub.PubSub` object. Higher up I would assert that publish and subscribe interfaces should be separate, and "third-party" libraries should be abstracted over, but low down it's not such a big deal.

Reply:
dont think its overkill, i think we should be doing more of this